### PR TITLE
Gerbil Update

### PR DESCRIPTION
**Pull request by fare**
###### Motivation for this change

Upstream changes plus new libraries.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/gambit/build.nix [hunk 1]
```diff
@@ -94,6 +95,7 @@ gccStdenv.mkDerivation rec {
 
     # Now use the bootstrap compiler to build the real thing!
     make -j$NIX_BUILD_CORES from-scratch
+    ${if gambit-params.modules then "make -j$NIX_BUILD_CORES modules" else ""}
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/gambit/build.nix:98]: Can you use optionalString?

Comment (fare) [reply pkgs/development/compilers/gambit/build.nix]: Done. In two places.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102936` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>gambit</li>
    <li>gambit-unstable</li>
    <li>gerbil</li>
    <li>gerbil-unstable</li>
    <li>go-libp2p-daemon</li>
  </ul>
</details>

**Comment by 7c6f434c** [comment]
A bit scared by Gambit apparently having a PHP translation target…

**Pull request merged**